ui/app/AppLayouts/Chat/panels/communities/JoinPermissionsOverlayPanel.qml

```diff
@@ -0,0 +1,100 @@
+import QtQuick 2.14
+import QtQuick.Controls 2.14
+import QtQuick.Layouts 1.14
+
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ.Components 0.1
+import StatusQ.Controls 0.1
+
+import AppLayouts.Chat.helpers 1.0
+import AppLayouts.Chat.controls.community 1.0
+
+import utils 1.0
+
+import SortFilterProxyModel 0.2
+
+
+Control {
+    id: root
+
+    property bool joinCommunity: true // Otherwise it means join channel action
+    property bool requirementsMet: true
+    property string communityName
+    property var communityHoldings
+    property string channelName
+    property var viewOnlyHoldings
+    property var viewAndPostHoldings
+    property var moderateHoldings
+
+    signal revealAddressClicked
+
+    QtObject {
+        id: d
+
+        readonly property string communityRequirementsNotMetText: qsTr("Membership requirements not met")
+        readonly property string communityRevealAddressText: qsTr("Reveal your address to join")
+        readonly property string channelRequirementsNotMetText: qsTr("Channel requirements not met")
+        readonly property string channelRevealAddressText: qsTr("Reveal your address to enter")
+
+        function holdingsTextFormat(name, amount) {
+            return CommunityPermissionsHelpers.setHoldingsTextFormat(HoldingTypes.Type.Asset, name, amount)
+        }
+    }
+
+    padding: 35 // default by design
+    spacing: 32 // default by design
+    implicitHeight: column.implicitHeight + topPadding + bottomPadding
+    implicitWidth: column.implicitWidth + leftPadding + rightPadding
+    contentItem: ColumnLayout {
+        id: column
```

Comment:
I believe width and height needs to specified in this case as a Layout is used (only no Layouts are automatically taking inherit the total size of their children eg Column or Row, ColunmLayout, RowLayout etc need instead a size to be set so that they can inherit to the children). Having it in contentItem, it seems that that ColumnLayout inherits the Control's dimensions which at the end are based on the ColumnLayout. Didn't any binding loop warning appear? did you try it with 5.15?

Reply:
I think that it's ok bc when control has size not explicitly set, it has size of it's content. When the size is provided for the control, then the content is adjusted accordingly. That's the biggest advantage of Control.

Reply:
yeah but the size provided for the control here is the content's size

Reply:
implicitHeight: column.implicitHeight and column has no implicitHeight set (same for width)

Reply:
This part:
```
    implicitHeight: column.implicitHeight + topPadding + bottomPadding
    implicitWidth: column.implicitWidth + leftPadding + rightPadding
```
is not needed here. Implicits are calculated according to that formula by default. I believe removing it clears all doubt here.

Reply:
Ok, I can remove these 2 lines so they are really redundant.